superset-frontend/src/dashboard/containers/FiltersBadge.tsx

```diff
@@ -38,16 +39,29 @@ const mapDispatchToProps = (dispatch: Dispatch) =>
   );
 
 const mapStateToProps = (
-  { datasources, dashboardFilters, charts }: any,
+  { datasources, dashboardFilters, nativeFilters, charts }: any,
   { chartId }: FiltersBadgeProps,
 ) => {
-  const indicators = selectIndicatorsForChart(
+  const dashboardIndicators = selectIndicatorsForChart(
     chartId,
     dashboardFilters,
     datasources,
     charts,
   );
 
+  const nativeIndicators = selectNativeIndicatorsForChart(nativeFilters);
+
+  const indicators = nativeIndicators.reduce((acc, indicator) => {
```

Comment:
I would suggest using `_.uniqBy()` here instead of `reduce()`.

Reply:
On second thought, you'd probably have to use `uniqWith`, not `uniqBy`.

Reply:
Thanks, that's a nice tip!